File: algorithms-step-by-step/recursion/exercises/palindrome.js
function isPalindrome(words) {
  let lengthOfString = words.length;
  for (let i = 0, j = lengthOfString - 1; i < lengthOfString; i++, j--) {
    if (words[i] !== words[j]) {
      return false;
    }
  }
  return true;
}

console.log(isPalindrome("awesome"));
console.log(isPalindrome("foobar"));
console.log(isPalindrome("tacocat"));
console.log(isPalindrome("amanaplanacanalpanama"));
console.log(isPalindrome("amanaplanacanalpandemonium"));

File: algorithms-step-by-step/recursion/exercises/palindrome.py
def is_palindrome(words):
    array = list(words)
    array.reverse()
    return ''.join(array) == words


print(is_palindrome('awesome'))
print(is_palindrome('foobar'))
print(is_palindrome('tacocat'))
print(is_palindrome('amanaplanacanalpanama'))
print(is_palindrome('amanaplanacanalpandemonium'))
